src/smtml/expr.ml

```diff
@@ -501,6 +516,26 @@ let cvtop ty op hte =
   | Ty.Cvtop.String_to_re, _ -> raw_cvtop ty op hte
   | _, Val v -> value (Eval.cvtop ty op v)
   | String_to_float, Cvtop (Ty_real, ToString, real) -> real
+  | Reinterpret_float, Cvtop (Ty_real, Reinterpret_int, e1) -> e1
+  | Reinterpret_float, Cvtop (Ty_fp 32, Reinterpret_int, e1) -> e1
+  | Reinterpret_float, Cvtop (Ty_fp 64, Reinterpret_int, e1) -> e1
+  | Reinterpret_int, Cvtop (Ty_int, Reinterpret_float, e1) -> e1
+  | Reinterpret_int, Cvtop (Ty_bitv 32, Reinterpret_float, e1) -> e1
+  | Reinterpret_int, Cvtop (Ty_bitv 64, Reinterpret_float, e1) -> e1
+  | ToString, Cvtop (_, OfString, e1) -> e1
+  | OfString, Cvtop (_, ToString, e1) -> e1
+  | ToBool, Cvtop (_, OfBool, e1) -> e1
+  | OfBool, Cvtop (_, ToBool, e1) -> e1
+  | PromoteF32, Cvtop (_, DemoteF64, e1) -> e1
```

Comment:
I am not sure if this is correct. When demoting the F64, you get a F32 where you lost some precision, and thus, the resulting F64 is not the same as the initial one?

Reply:
Yes, these two I'm not sure the round-trip is safe